hardware_interface/test/test_component_interfaces.cpp

```diff
@@ -0,0 +1,577 @@
+// Copyright 2020 ros2_control development team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "hardware_interface/actuator_hardware.hpp"
+#include "hardware_interface/actuator_hardware_interface.hpp"
+#include "hardware_interface/component_info.hpp"
+#include "hardware_interface/hardware_info.hpp"
+#include "hardware_interface/joint.hpp"
+#include "hardware_interface/sensor_hardware_interface.hpp"
+#include "hardware_interface/sensor_hardware.hpp"
+#include "hardware_interface/sensor.hpp"
+#include "hardware_interface/system_hardware_interface.hpp"
+#include "hardware_interface/system_hardware.hpp"
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+#include "hardware_interface/types/hardware_interface_status_values.hpp"
+
+using namespace ::testing;  // NOLINT
+
+namespace hardware_interface
+{
+
+namespace hardware_interfaces_components_test
+{
+
+class DummyPositionJoint : public Joint
+{
+public:
+  return_type configure(const ComponentInfo & joint_info) override
+  {
+    info = joint_info;
+    if (info.command_interfaces.size() > 1 || info.state_interfaces.size() > 1) {
+      return return_type::ERROR;
+    }
+    if (info.command_interfaces.size() == 0) {
+      info.command_interfaces.push_back("position");
```

Comment:
shouldn't this use one of the constants defined in hardware_interface_type_values.hpp?